epochStart/bootstrap/process_test.go

```diff
@@ -575,74 +1102,449 @@ func TestSyncUserAccountsState(t *testing.T) {
+
+		prevShardHeader := &block.Header{}
+		notarizedShardHeader := &block.Header{
+			PrevHash: prevShardHeaderHash,
+		}
+		metaBlock := &block.MetaBlock{
+			Epoch: 2,
+			EpochStart: block.EpochStart{
+				LastFinalizedHeaders: []block.EpochStartShardData{
+					{HeaderHash: notarizedShardHeaderHash, ShardID: 0},
+				},
+			},
+		}
+
+		epochStartProvider, _ := NewEpochStartBootstrap(args)
+		epochStartProvider.epochStartMeta = metaBlock
+		epochStartProvider.headersSyncer = &epochStartMocks.HeadersByHashSyncerStub{
+			GetHeadersCalled: func() (m map[string]data.HeaderHandler, err error) {
+				return map[string]data.HeaderHandler{
+					string(notarizedShardHeaderHash): notarizedShardHeader,
+					string(prevShardHeaderHash):      prevShardHeader,
+				}, nil
+			},
+		}
+		epochStartProvider.dataPool = dataRetrieverMock.NewPoolsHolderMock()
+
+		epochStartProvider.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{}
+
+		err := epochStartProvider.requestAndProcessForMeta()
+		assert.Equal(t, state.ErrNilRequestHandler, err)
+	})
+	t.Run("fail to sync user accounts state", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEpochStartBootstrapArgs(coreComp, cryptoComp)
+
+		prevShardHeader := &block.Header{}
+		notarizedShardHeader := &block.Header{
+			PrevHash: prevShardHeaderHash,
+		}
+		metaBlock := &block.MetaBlock{
+			Epoch: 2,
+			EpochStart: block.EpochStart{
+				LastFinalizedHeaders: []block.EpochStartShardData{
+					{HeaderHash: notarizedShardHeaderHash, ShardID: 0},
+				},
+			},
+		}
+
+		epochStartProvider, _ := NewEpochStartBootstrap(args)
+		epochStartProvider.syncedHeaders = make(map[string]data.HeaderHandler)
+		epochStartProvider.epochStartMeta = metaBlock
+		epochStartProvider.headersSyncer = &epochStartMocks.HeadersByHashSyncerStub{
+			GetHeadersCalled: func() (m map[string]data.HeaderHandler, err error) {
+				return map[string]data.HeaderHandler{
+					string(notarizedShardHeaderHash): notarizedShardHeader,
+					string(prevShardHeaderHash):      prevShardHeader,
+				}, nil
+			},
+		}
+		epochStartProvider.dataPool = &dataRetrieverMock.PoolsHolderStub{
+			TrieNodesCalled: func() storage.Cacher {
+				return &testscommon.CacherStub{
+					GetCalled: func(key []byte) (value interface{}, ok bool) {
+						return nil, true
+					},
+				}
+			},
+		}
+
+		epochStartProvider.miniBlocksSyncer = &epochStartMocks.PendingMiniBlockSyncHandlerStub{}
+		epochStartProvider.requestHandler = &testscommon.RequestHandlerStub{}
+
+		err := epochStartProvider.requestAndProcessForMeta()
+		assert.Error(t, err)
+	})
+}
+
+func TestPrepareComponentsToSyncFromNetwork(t *testing.T) {
+	t.Parallel()
+
+	coreComp, cryptoComp := createComponentsForEpochStart()
+	args := createMockEpochStartBootstrapArgs(coreComp, cryptoComp)
 
 	shardCoordinator := mock.NewMultipleShardsCoordinatorMock()
 	shardCoordinator.CurrentShard = 0
 
 	epochStartProvider, _ := NewEpochStartBootstrap(args)
```

Comment:
[question]

Why after every `NewEpochStartBootstrap` with args you get from `createMockEpochStartBootstrapArgs` you need to set internal fields "manually"? Couldn't they be set by default in constructor by using other "valid" args?

Reply:
actually i just followed the setup that was already in place; `epochStartBoostrap` structure has a set of fields which are created/setup along the the way, setting everything in the constructor would require a pretty big refactoring, this might be considered in a future PR